compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -401,6 +401,7 @@ object TypeOps:
   def avoid(tp: Type, symsToAvoid: => List[Symbol])(using Context): Type = {
     val widenMap = new ApproximatingTypeMap {
       @threadUnsafe lazy val forbidden = symsToAvoid.toSet
+      val locals = util.HashSet[Type]()
```

Comment:
As an optimization, this could be an uninitialized var which is only initialized the first time we encounter an HKTypeLambda.

Reply:
(or just make it a lazy val).